daemon/api_snapshots_test.go

```diff
@@ -68,6 +72,48 @@ func (s *snapshotSuite) TestSnapshotMany(c *check.C) {
 	c.Check(res.Affected, check.DeepEquals, inst.Snaps)
 }
 
+func (s *snapshotSuite) TestSnapshotManyOptionsFull(c *check.C) {
+	defer daemon.MockSnapshotSave(func(s *state.State, snaps, users []string,
```

Comment:
(super nitpick) I generally like in these mocked helpers to have a counter how often they are called, it's really not strictly needed here but it's a nice pattern to keep in mind so I wanted to mention it. Something like:
```go
diff --git a/daemon/api_snapshots_test.go b/daemon/api_snapshots_test.go
index afd89845c4..d47bacf505 100644
--- a/daemon/api_snapshots_test.go
+++ b/daemon/api_snapshots_test.go
@@ -73,8 +73,10 @@ func (s *snapshotSuite) TestSnapshotManyOptionsNone(c *check.C) {
 }
 
 func (s *snapshotSuite) TestSnapshotManyOptionsFull(c *check.C) {
+       var snapshotSaveCalled int
        defer daemon.MockSnapshotSave(func(s *state.State, snaps, users []string,
                options map[string]*snap.SnapshotOptions) (uint64, []string, *state.TaskSet, error) {
+               snapshotSaveCalled++
                c.Check(snaps, check.HasLen, 2)
                c.Check(options, check.HasLen, 2)
                c.Check(options, check.DeepEquals, map[string]*snap.SnapshotOptions{
@@ -95,6 +97,7 @@ func (s *snapshotSuite) TestSnapshotManyOptionsFull(c *check.C) {
        c.Assert(err, check.IsNil)
        c.Check(res.Summary, check.Equals, `Snapshot snaps "foo", "bar"`)
        c.Check(res.Affected, check.DeepEquals, inst.Snaps)
+       c.Check(snapshotSaveCalled, check.Equals, 1)
 }
 
 func (s *snapshotSuite) TestSnapshotManyError(c *check.C) {
```
I like it because it ensure that the test and code behaves correctly, the right mock is called for exactly the right times. Not a problem here but in the past there were tests where accidently the mock was not called at all but because the test setup did other mocking the test passed but the checks inside the mock never got called so the test looked great but did actually test less than what it looked like (again, not really an issue here as the res checks are pretty specific).

Reply:
Very useful thanks. Added here and 1 other place. Will adopt this going forward.